src/calculate-totals.js

```diff
@@ -0,0 +1,148 @@
+const calculateTotals = (result) => {
+  if (result.data.length === 0) {
+    return result
+  }
+
+  let totals = {}
+
+  // Sum up simple columns
+  if ("users" in result.data[0]) {
+    totals.users = _sumColumn({ column: "users", result })
+  }
+  if ("visits" in result.data[0]) {
+    totals.visits = _sumColumn({ column: "visits", result })
+  }
+
+  // Sum up categories
+  if (result.name.match(/^device_model/)) {
+    totals.device_models = _sumVisitsByColumn({
+      column: "mobile_device",
+      result,
+    })
+  }
+  if (result.name.match(/^language/)) {
```

Comment:
I think it was probably a premature optimization for future-proofing, or perhaps to make it easier for other users to create similar report names without going back to the logic. But either way, this doesn't make sense without it being a Documented Supported Feature, which it isn't. :+1: to clean it up for precision.